namer/core/src/test/scala/io/buoyant/namer/NamerTestUtil.scala

```diff
@@ -0,0 +1,26 @@
+package io.buoyant.namer
+
+import com.twitter.finagle.{Name, NameTree, Namer, Path}
+import com.twitter.util.{Await, Future}
+import org.scalatest.FunSuite
+
+trait NamerTestUtil { self: FunSuite =>
+
+  protected def assertBoundIdAutobinds(namer: Namer, path: Path, prefix: Path): Unit = {
+    def lookup(p: Path) = {
+      assert(p.startsWith(prefix))
+      val pathWithoutPrefix = p.drop(prefix.size)
+      Await.result(namer.lookup(pathWithoutPrefix).values.toFuture.flatMap(Future.const))
```

Comment:
that would work in all the tests so far, but would fail for any Activity that starts out as pending.